From 77fc470bbb3ecc19fb8d1058985c7b591235aeb3 Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Fri, 28 Oct 2011 19:15:42 +0800
Subject: [PATCH 044/641] UniCore64: Add some functions about thread

Fixed error:
noinitramfs.c:(.ref.text+0x28): undefined reference to `kernel_thread'
elfcore.c:(.text+0x7ab8): undefined reference to `copy_thread'
elfcore.c:(.text+0xe8fc): undefined reference to `exit_thread'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/process.c |   38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/unicore64/kernel/process.c b/arch/unicore64/kernel/process.c
index 1e44fd1..3b436ef 100644
--- a/arch/unicore64/kernel/process.c
+++ b/arch/unicore64/kernel/process.c
@@ -1,6 +1,44 @@
 #include <linux/kernel.h>
 
 /**
+ * kernel_thread() - Create a kernel thread
+ * @fn:
+ * @arg:
+ * @flags:
+ */
+int kernel_thread(int (*fn)(void *), void * arg, unsigned long flags)
+{
+	/* FIXME */
+	BUG();
+	return 0;
+}
+
+/**
+ * copy_thread() - Copy a thread
+ * @clone_flags:
+ * @stack_start:
+ * @stk_sz:
+ * @p:
+ * @regs:
+ */
+int copy_thread(unsigned long clone_flags, unsigned long stack_start,
+	    unsigned long stk_sz, struct task_struct *p, struct pt_regs *regs)
+{
+	/* FIXME */
+	BUG();
+	return 0;
+}
+
+/**
+ * exit_thread() - Free current thread data structures etc..
+ */
+void exit_thread(void)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
  * machine_restart() - Reboot the machine
  * @cmd:
  */
-- 
1.7.9.5

